### DISCUSSION TITLE
Docker Registry Install creates new certificate on each run

### DISCUSSION THREAD

**Discussion by stoeps13** [HCL-TECH-SOFTWARE/connections-automation#80] [Q&A] [answered]
Using the playbook to install the Docker registry and you deploy on multiple k8s_workers there are several issues:

- Folder on worker nodes for ca.crt is wrong, because name is set to hostname and not docker registry
- ca.crt is not copied to worker_nodes
- rerunning the playbook generates new certificates, so even manually adjusted certificates on the workers do not work any more

I think this needs some adjustments, or did I miss something? I use https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/d097b3274a8825657618147facf2372fe763b41d/playbooks/setup-component-pack-complete.yml (but had similar issues with the other playbooks too).

**Comment by dejo1307** [comment, marked as answer]
I'm not sure that I understand what is the issue? I'm not sure if something changed, but this was done as per official documentation for setting up Docker Registry.

If I remember correctly, it is all just fine if you have a single node, but if you have multiple nodes, since you are creating self signed certificate, there is the process where that certificate is made trusted on all the nodes so docker login can work in the end. Are you saying that this doesn't work?

Regarding certificates recreation, I'm failing to see why this is a problem? This is exactly what we wanted to happen, as we catch at least three times in a year this way that process changed and that we had to update the automation.

Can you elaborate a bit more on the problem here?

**Reply by stoeps13** [reply]
Hi Dejan,
yes in my tests the trust on the worker nodes was not copied. First the folder is wrong, because it creates /etc/docker/certs.d/hostname:5000/ca.crt and not the registry host! And even with the folder, no ca.crt is copied to it. So no pod can be deployed (starting with bootstrap, because the registry is not trusted).

Recreating is ok, but it needs time and I just don't like it from a idempotency approach. When the cert is valid it should not be renewed.

**Reply by stoeps13** [reply]
Ok, found my mistake, I need to add all k8s_workers to docker_registry group. Thanks   @dejo1307
